### PR TITLE
init selftests: output status tests

### PR DESCRIPTION
**Pull request by dannasessha**
I followed up on notes in PR #3: this is based off the current `main` branch before #2 was merged in.

Included here are a set of basic integration tests, found in `mod selftests`. They test the output of `checkmate --help` and the success exit code of that cli command, along with the success of 5 of the 7 phases used by `checkmate`. The other phases (`audit` and `test`) do not pass using the same logic. There is a single negtive control for an argument not offered by the cli documentation. 

Three other arguments exist in the cli documentation, which I have not yet addressed: 
`everything`, `hook`, `readme`.

`tarpaulin` seems to only offer statistics when all tests pass, so we can use that as a metric (this commit shows `25.71% coverage, 99/385 lines covered`) But I think creating a test to show a failure with `structopt` and then a success with `clap` would have to work with or around this limitation (I'd imagine it could be added at the last moment).

### CODE REVIEW

**Review by AloeareV — APPROVED** [review]
All looks good!

File: tests/selftests.rs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        assert!(super::common::checkmate_command("doc").status.success())
+    }
+
+    #[test]
+    fn checkmate_build_status_successful() {
+        assert!(super::common::checkmate_command("build").status.success())
+    }
+
+    #[test]
+    fn checkmate_foo_status_fails() {
```
Comment (zancas) [inline tests/selftests.rs:59]: Nice.

File: tests/selftests.rs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    #[test]
+    fn checkmate_foo_stderr_contains_specific_response() {
+        let s_err = super::common::checkmate_command("foo").stderr;
+        let err = str::from_utf8(&s_err).unwrap();
+        assert_eq!(
+            true,
+            err.contains(
+                "error: Found argument 'foo' which wasn't expected, or isn't valid in this context"
```
Comment (zancas) [inline tests/selftests.rs:73]: Nice!

**Review by zancas — APPROVED** [review]

**Comment by zancas** [comment]
Hey @dannasessha I am +1 on merge.   There's a conflict which you should be able to trivially resolve in-browser with the "Resolve conflicts" button.

Over all this looks really thorough.

**Comment by dannasessha** [comment]
tarpaulin reports `28.50% coverage, 114/400 lines covered` on latest commit.

File: tests/common.rs [hunk 3]
```diff
@@ -1,25 +1,9 @@
 use std::process::Command;
 use std::process::Output;
 
-pub fn assemble_command(argument: &str) -> Output {
-    Command::new("sh")
-        .arg("-c")
+pub fn checkmate_command(argument: &str) -> Output {
```
Comment (zancas) [inline tests/common.rs:4]: Should this `fn` require `arg` to be non-empty?

**Comment by dannasessha** [comment]
Force push due to operator error, should be corrected now.

File: tests/common.rs (hunk 3 above)
Comment (dannasessha) [reply tests/common.rs:4]: If the (required) argument is `""` then the command should only be appended by a space. I don't think that would change any behavior from a non-argument case (of which there are none currently included in the tests).

This `fn` is only called from the tests at this time, but I could add a test or a case for this.

**Pull request merged**